diff --git a/default/be_modtab.c b/default/be_modtab.c
--- a/default/be_modtab.c
+++ b/default/be_modtab.c
@@ -8,0 +9,2 @@
+#include <nuttx/config.h>
+
@@ -25,0 +28,3 @@
+#ifdef CONFIG_INTERPRETERS_BERRY_LVGL
+be_extern_native_module(lv);
+#endif
@@ -70,0 +76,3 @@
+#ifdef CONFIG_INTERPRETERS_BERRY_LVGL
+    &be_native_module(lv),
+#endif
diff --git a/default/berry.c b/default/berry.c
--- a/default/berry.c
+++ b/default/berry.c
@@ -10,0 +11 @@
+#include <nuttx/config.h>
@@ -13,0 +15,4 @@
+
+#ifdef CONFIG_INTERPRETERS_BERRY_LVGL
+extern void be_lvgl_cleanup(bvm *vm);
+#endif
@@ -22 +27,3 @@
-#if defined(__linux)
+#if defined(__NuttX__)
+    #define OS_NAME   "NuttX"
+#elif defined(__linux)
@@ -101 +108 @@
-    const char *optarg;
+    const char *optvalue;
@@ -146 +153 @@
-                opt->optarg = argv[opt->idx++]; /* save the argument */
+                opt->optvalue = argv[opt->idx++]; /* save the argument */
@@ -149 +156 @@
-            opt->optarg = NULL;
+            opt->optvalue = NULL;
@@ -270 +277 @@
-            opt->modulepath = opt->optarg;
+            opt->modulepath = opt->optvalue;
@@ -275 +282 @@
-            opt->src = opt->optarg;
+            opt->src = opt->optvalue;
@@ -279 +286 @@
-            opt->dst = opt->optarg;
+            opt->dst = opt->optvalue;
@@ -333,0 +341,10 @@
+
+#ifdef CONFIG_INTERPRETERS_BERRY_LVGL_GLOBAL
+static void berry_preload_lvgl(bvm *vm)
+{
+    if (be_getmodule(vm, "lv")) {
+        be_setglobal(vm, "lv");
+        be_pop(vm, 1);
+    }
+}
+#endif
@@ -403,0 +422,3 @@
+#ifdef CONFIG_INTERPRETERS_BERRY_LVGL_GLOBAL
+    berry_preload_lvgl(vm);
+#endif
@@ -404,0 +426,3 @@
+#ifdef CONFIG_INTERPRETERS_BERRY_LVGL
+    be_lvgl_cleanup(vm);
+#endif
